3_summarize/src/annual_thermal_metrics.R

```diff
@@ -365,6 +367,29 @@ calc_first_day_above_temp <- function(date, wtr_surf, temperatures) {
   return(date_above_df)
 }
 
+#' date where surface temperature drops below a certain temperature in deg C
+#' after the summer peak; returns the Julian day
+calc_first_day_below_temp <- function(date, wtr_surf, temperatures, peak_temp_dt) {
+  
+  # Need one date & wtr_surf per day
+  date_unique <- unique_day(date)
+  wtr_surf_unique <- unique_day_data(date, wtr_surf)
+  
+  # Only keep dates after peak summer date because we want to know the fall date specifically
+  post_summer_peak_i <- which(date_unique > peak_temp_dt)
+  dates_post_summer <- date_unique[post_summer_peak_i]
+  water_surf_post_summer <- wtr_surf_unique[post_summer_peak_i]
+  
+  date_below_df <- lapply(temperatures, function(temp) {
+    first_day_below_temp <- dates_post_summer[water_surf_post_summer < temp] %>% head(1) %>% format("%j") %>% as.numeric()
+    if(length(first_day_below_temp) == 0) first_day_below_temp <- NA # Return NA if that temp was never reached
+    return(first_day_below_temp)
+  }) %>% data.frame()
+  names(date_below_df) <- sprintf("date_below_%s", temperatures)
```

Comment:
Am assuming this operates on yearly data and the year or date column isn't necessary to return since you've got a larger function calling all of these sub-functions and the parent function has the year information in the `data.frame` you are cbinding (or something else) to.

Reply:
yes correct. This is a function called within the `summarize` part of a `group_by(year) %>% summarize()`